test/mediastream.js

```diff
@@ -75,6 +75,13 @@ tape('new MediaStream(tracks)', function(t) {
   });
 });
 
+tape('new MediaStream(mediaStreamInit)', function(t) {
+  var stream = new MediaStream({ id: 'testStreamId' });
```

Comment:
Ca you please include a test to see what happens when two MediaStreams with the same ID are created? Also, I'm curious what happens if two MediaStreams with the same ID are used on an RTCPeerConnection.

Reply:
Sorry I should have included some details on this.  I did set out to do this, however I backed it out.  Here's what I did:
-I analyzed the webrtc.org code base for what would happen if 2 MediaStreams had the same id.  I discovered that there was no major concern here.  It appears that worse case the 2nd stream would fail to get added to a peer connection if one with the same name already existed.
-I thought about adding a test case to test/mediastream.js - however this file really just deals with MediaStream objects in isolation.  So even though I could create test case where we create 2 media streams with the same id, we wouldn't really be testing anything useful, until they are added to a track or transceiver.
-I then set out to add such test to test/rtcrtpsender.js (since it has addTrack tests in it), however I was unable to come up with any meaningful way to verify anything (ie: assert conditions), since I can't actually retrieve the stream id, from an interface outside the MediaStream object itself, once it is set.  Also - we know that addTrack and addTransceiver in the C++ side don't really do anything with the media stream object, other than extracting the id from it.

If you have some other idea's or approaches I could use, I would be happy to try to implement them.  :)

Scott

Reply:
OK, I found a way to verify things by looking at the generated SDP.  I added a new test case to rtcrtpsender.js - let me know what you think.  :)